### ISSUE TITLE
Types

### ISSUE THREAD

**Issue by valguss** [Skywire/m2-module-wordpress-api#26]
Rather than having to code in each type that wordpress uses (post,page etc) use the types API to get a list of types: https://developer.wordpress.org/rest-api/reference/post-types/

**Comment by Demoli** [comment]
Compile index / cache of all URLs by type (post, category, page, custom types (style-genius)).
When routing check this index, if URL doesn't exist, return 404.

This prevents us spamming WP with 404s when a route doesn't exist.

**Comment by Demoli** [comment]
https://github.com/Skywire/m2-module-wordpress-api/pull/28

The types API was returning things we don't want, and not returning things we do want, so my indexer uses DI to query the types instead.
```
<type name="\Skywire\WordpressApi\Model\Index\Spider">
        <arguments>
            <argument name="types" xsi:type="array">
                <item name="categories" xsi:type="string">categories</item>
                <item name="posts" xsi:type="string">posts</item>
                <item name="pages" xsi:type="string">pages</item>
            </argument>
        </arguments>
    </type>
```

In any custom type modules you'll just need to add a new type via the DI, and it will be indexed like the others.

Indexing is on a nightly cron, it empties and repopulates the index each night, there's no TTL or diff behaviour at the moment.

The router now has a plugin that will check if the slug exists in the index table, if it doesn't the router will return false and no requests will be made to WP.

If it does exist the current router behaviour will apply.

**Comment by Demoli** [comment]
Suggest we try this new version on a new build and then backport it.

**Comment by Demoli** [comment]
fix #28
